File: src/commands/ShowCommand.java
package commands;

import commands.core.ServerSideCommand;
import lib.ExecutionStatus;
import lib.functions.CallbackFunction;
import organization.OrganizationManager;

import java.io.IOException;

public class ShowCommand extends ServerSideCommand {

    public ShowCommand(CallbackFunction callbackFunction, OrganizationManager organizationManager) {
        super(callbackFunction, organizationManager);
    }

    @Override
    public void executeImplementation(String[] args, CallbackFunction callback) throws IOException {
        String mode = args.length == 0 ? "yaml" : args[0].toLowerCase();
        String result = switch (mode) {
            case "json" -> organizationManager.toJson();
            case "csv" -> organizationManager.toCSV();
            case "yaml" -> organizationManager.toYaml();
            default -> null;
        };

        callback.invoke(result == null ? ExecutionStatus.FAILURE : ExecutionStatus.SUCCESS, null, result);
    }
}

File: src/commands/client_side/ShowCommand.kt
package commands.client_side

import commands.client_side.core.ServerSideCommand
import database.OrganizationManagerInterface
import exceptions.InvalidOutputFormatException

class ShowCommand(organizationDatabase: OrganizationManagerInterface) :
    ServerSideCommand(organizationDatabase) {
    override fun executeImplementation(argument: Any?): Result<String> {
        val mode = argument as String?

        return when (mode) {
            null, "yaml" -> Result.success(organizationDatabase.toYaml())
            "json" -> Result.success(organizationDatabase.toJson())
            "csv" -> Result.success(organizationDatabase.toCSV())
            else -> Result.failure(InvalidOutputFormatException())
        }
    }
}
